src/sage/graphs/cographs.py

```diff
@@ -0,0 +1,529 @@
+r"""
+Cographs
+
+A cograph is a `P_4`-free graph, that is a graph without induced path of order
+4. Any cograph may be constructed, starting from the single vertex graph, by a
+sequence of join and disjoint union operations. See the :wikipedia:`Cograph` for
+more details on this graph class, and :oeis:`A000084` to know the number of
+cographs of order `n \geq 1`.
+
+This module implements the folowing methods concerning cographs:
+
+.. csv-table::
+    :class: contentstable
+    :widths: 30, 70
+    :delim: |
+
+    :meth:`cographs` | Return an iterator over the cographs of order `n`.
+
+Methods
+-------
+"""
+# ****************************************************************************
+#       Copyright (C) 2017-2024 Marianna Spyrakou
+#                          2024 David Coudert <david.coudert@inria.fr>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.combinat.partitions import AccelAsc_next
+
+
+class CoTree:
+    r"""
+    Generic cotree node.
+
+    This data structure is used for the generation of cographs in
+    :meth:`cographs`.
+    """
+    def __init__(self, name='root'):
+        r"""
+        Initialize a cotree.
+
+        INPUT:
+
+        - ``name`` -- either an operation ('U' or 'J') or the size of the
+          subtree rooted at this node
+
+        EXAMPLES::
+
+            sage: from sage.graphs.cographs import CoTree
+            sage: CoTree(1)
+            ( 1 )
+            sage: CoTree()
+            ( root )
+        """
+        self.name = name
+        self.children = []
+        self.info = None
+        self.parent = None
+
+    def __repr__(self):
+        r"""
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+            sage: from sage.graphs.cographs import CoTree
+            sage: CoTree(1)
+            ( 1 )
+            sage: CoTree('J')
+            [ J ]
+            sage: next(graphs.cographs(4, as_graph=False))  # indirect doctest
+            [ J ( 0 ) ( 1 ) ( 2 ) ( 3 ) ]
+        """
+        return self.__str__()
```

Comment:
```suggestion
    __repr__ = __str__
```
I believe for Python classes the dunder methods are fine to be aliases like this (but this will have to be after the `__str__` declaration).